src/libraries/System.Console/src/System/ConsolePal.iOS.cs

```diff
@@ -0,0 +1,201 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+using System.Text;
+using System.Runtime.InteropServices;
+
+namespace System
+{
+    internal class NSLogStream : Stream
+    {
+        private bool _error;
+
+        public NSLogStream(bool error)
+        {
+            _error = error;
+        }
+
+        public override void Flush() { }
+
+        public override int Read(byte[] buffer, int offset, int count) => throw new PlatformNotSupportedException();
+
+        public override long Seek(long offset, SeekOrigin origin) => throw new PlatformNotSupportedException();
+
+        public override void SetLength(long value) => throw new PlatformNotSupportedException();
+
+        // call NSLog
```

Comment:
Why can't we call NSLog directly?

Reply:
The rule for code under libraries is to avoid depending on runtime implementation details. If this needs PAL, it should go to System.Native.

Reply:
Yep this needs to go interop layer, probably under osx as there is no better name for now.

Reply:
It looks like it's a bit more complicated than that - NSLog requires an instance of NSString, it's also have different signatures between armv7 and arm64 and watchOS requires a different approach, so I basically re-used the Xamarin.iOS logic and moved it to System.Native in my latest commit